### PR TITLE
Attempt to fix race condition in worker/caasunitprovisioner

### PR DESCRIPTION
**Pull request by benhoyt**
I think what was happening was occasionally the parent worker was getting another event that would cause it to remove the
applicationWorker before the test expected it, and the `c.Assert(ok, jc.IsTrue)` would fail (or previously panic when the assert
wasn't there).

Now we fetch the applicationWorker earlier in the test when we know it should be active, so the `CheckKilled` argument will never be nil.

This was causing a panic in our race tests ([example](https://jenkins.juju.canonical.com/job/Unit-RunUnitTests-race-amd64/1133/console), also copied below), then @wallyworld changed it to fail instead of panic with the assert. Hopefully this fixes the intermittent failure case.

```
06:31:51 panic: runtime error: invalid memory address or nil pointer dereference
06:31:51 [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x223c44a]
06:31:51 
06:31:51 goroutine 313 [running]:
06:31:51 github.com/juju/juju/worker/caasunitprovisioner.(*applicationWorker).Wait(0x0)
06:31:51 	/home/jenkins/workspace/Unit-RunUnitTests-race-amd64/_build/src/github.com/juju/juju/worker/caasunitprovisioner/application_worker.go:81 +0x2a
06:31:51 github.com/juju/worker/v3/workertest.CheckKilled.func1()
06:31:51 	/home/jenkins/workspace/Unit-RunUnitTests-race-amd64/_build/src/github.com/juju/juju/vendor/github.com/juju/worker/v3/workertest/check.go:52 +0x49
06:31:51 created by github.com/juju/worker/v3/workertest.CheckKilled
06:31:51 	/home/jenkins/workspace/Unit-RunUnitTests-race-amd64/_build/src/github.com/juju/juju/vendor/github.com/juju/worker/v3/workertest/check.go:51 +0x12c
06:31:51 FAIL	github.com/juju/juju/worker/caasunitprovisioner	1.289s
```

### CODE REVIEW

File: worker/caasunitprovisioner/worker_test.go [hunk 1]
```diff
@@ -1018,9 +1018,12 @@ func (s *WorkerSuite) TestV2CharmExitsApplicationWorker(c *gc.C) {
 	s.sendApplicationChanges(c, "gitlab")
 	waitCharmGetterCalls("ApplicationCharmInfo")
 
-	// Wait till Life() is called to synchronize (ensure charmInfo is done).
+	// Wait till application worker has started up
+	var appWorker worker.Worker
 	for a := coretesting.LongAttempt.Start(); a.Next(); {
-		if len(s.lifeGetter.Calls()) > 0 {
+		aw, ok := caasunitprovisioner.AppWorker(w, "gitlab")
```
Comment (wallyworld) [inline worker/caasunitprovisioner/worker_test.go:1024]: We need to assert that appWorker is set after we exit the loop

**Review by wallyworld — APPROVED** [review]
Just one small change

File: worker/caasunitprovisioner/worker_test.go (hunk 1 above)
Comment (benhoyt) [reply worker/caasunitprovisioner/worker_test.go:1024]: Ah, thanks - fixed.

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**